src/intrprtr.c

```diff
@@ -2924,12 +2924,10 @@ void            IntrAssDVar (
     rhs = PopObj();
 
     /* assign the right hand side                                          */
-    currLVars = STATE(CurrLVars);
-    SWITCH_TO_OLD_LVARS( STATE(ErrorLVars) );
+    context = STATE(ErrorLVars);
     while (depth--)
-      SWITCH_TO_OLD_LVARS( PARENT_LVARS(STATE(CurrLVars)) );
-    ASS_HVAR( dvar, rhs );
-    SWITCH_TO_OLD_LVARS( currLVars  );
+      context = PARENT_LVARS(STATE(CurrLVars));
```

Comment:
Oops, this is wrong -- `CurrLVars ` should be `context`. I'll push a revised PR.

I'll see if I can add a test for this, too....